packages/react-router-dev/vite/plugin.ts

```diff
@@ -3276,6 +3323,11 @@ export async function getEnvironmentOptionsResolvers(
         ssrEmitAssets: true,
         copyPublicDir: false, // Assets in the public directory are only used by the client
         rollupOptions: {
+          input:
+            (ctx.reactRouterConfig.future.unstable_viteEnvironmentApi
+              ? viteUserConfig.environments?.ssr?.build?.rollupOptions?.input
+              : viteUserConfig.build?.rollupOptions?.input) ??
+            virtual.serverBuild.id,
```

Comment:
Since the `route-ids` query string is no longer used, this config is now common across all SSR builds including server bundles.